src/dlgTriggerEditor.cpp

```diff
@@ -6718,6 +6734,212 @@ void dlgTriggerEditor::exportKey( QFile & file )
 
 }
 
+void dlgTriggerEditor::exportTriggerToClipboard( )
+{
+    QString name;
+    TTrigger * pT = 0;
+    QTreeWidgetItem * pItem = treeWidget_triggers->currentItem();
+    if( pItem )
+    {
+        int triggerID = pItem->data( 0, Qt::UserRole ).toInt();
+        pT = mpHost->getTriggerUnit()->getTrigger( triggerID );
+        if( pT )
+        {
+            name = pT->getName();
+        }
+        else
+        {
+            QMessageBox::warning(this, tr("Export Package:"),
```

Comment:
Well I think such ___horrible___, ___evil___ constructs (is that how you see them Vadim?):see_no_evil: are also used if nothing is selected and the export button is used! This example is from my current WIP but shows what happens if you hit the _Export_ button without selecting anything first.  Note you do have to select a file to save the export as a XML file first before you get this warning in the "Export" case:
![trying to export no item qmessagebox](https://cloud.githubusercontent.com/assets/6163092/23675466/66bee86c-0371-11e7-8ccf-c4e78bcc4f5f.png)
We should not blame @dicene for repeating this behaviour so perhaps we will have to allow it **and review several uses of `QMessageBox`** instances in the editor - there are some that ARE justifiable IMHO.  Some coders may not be aware that the predefined simpler `static` members being used here are `Qt::ApplicationModal` which means they block input to all other application windows until closed...